backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/course/domain/CourseExecution.java

```diff
@@ -222,4 +236,17 @@ public int getNumberOfQuestions() {
                 .filter(user -> user.getRole().equals(User.Role.STUDENT))
                 .collect(Collectors.toSet());
     }
+
+    public Set<Topic> findAvailableTopics() {
+        return getAvailableAssessments().stream().flatMap(assessment -> assessment.getTopics().stream()).distinct().collect(Collectors.toSet());
```

Comment:
Since it returns a set we can remove the distinct

Reply:
Done